shell/components/form/Error.vue

```diff
@@ -0,0 +1,37 @@
+<script>
+export default {
+  props: {
+    value: {
+      default: null,
+      type:    Object
+    },
+    rules: {
+      default:   () => [],
+      type:      Array,
+      // we only want functions in the rules array
+      validator: rules => rules.every(rule => ['function'].includes(typeof rule))
+    }
+  },
+  data() {
+    return {};
+  },
+  computed: {
+    validationMessage() {
```

Comment:
Could you please make a test out of this?

As far as I can see we are pushing the result of a function, which can be pretty much anything.
I would recommend to add specificity with a type.
You must consider also that if you may return `undefined` or empty string, you may end up with results as the following:
```ts
[undefined, ''].join(', ')      //      ', '
```